docs/DevGuide/Variables.md

```diff
@@ -0,0 +1,75 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Using Variables in SpECTRE
+
+# What is a Variables and Why Use Them?
+Variables are a data structure that allow you to make most if not all memory
+allocations needed at one time. Allocating memory can be expensive as it holds
+up any computations while the memory is being allocated. Using a Variables to
+allocate memory before it's needed allows the computation to operate smoothly
+uninteruppted.
+
+# Defining a Variables
+To define a Variables, you'll need to
+`#include "DataStructures/Tags/TempTensor.hpp"` this will give you access to
+temporary Scalars and Tensors we'll need to allocate. Once you have that,
+you can define a Variables that allocates one Scalar like this
+```
+Variables<tmpl::list<::Tags::TempScalar<0>>>
+temp_buffer{get<0,0>(spatial_metric).size()};
+```
+Here, the Variables we've defined makes a temp_buffer that will be used to
```

Comment:
This is a personal preference thing, so up to you! I putting class names and variable names in a single tick block: `Variables` or `temp_buffer`